narayana/scripts/hudson/benchmark.sh

```diff
@@ -1,4 +1,4 @@
-[ -z "${JMHARGS}" ] && JMHARGS="-t 240 -r 25 -f 2 -wi 5 -i 5"
+[ -z "${JMHARGS}" ] && JMHARGS="-t 240 -r 30 -f 3 -wi 5 -i 5"
```

Comment:
The issue is asking to either:

1. Reduce the number of forks (from 10 to 3), or
2. Run a subset of the performance tests on each PR and then run them all once a week

Since the number of forks was reduced in a previous PR (unless the fork count has been overridden in the CI job) and has had little to no effect on the run to run variance I suggest that this PR just implements option 2.

Note that we run a weekly performance job (look for a job called narayana-performance in the Jenkins console). This satisfies the requirement to run all of the performance tests weekly. And to just run a subset of the performance regression tests against a pull request you will need to modify the bash script that runs the regression tests. In detail:

> If a PR has `PERF` in the description then our Jenkins CI will run the PERF profile (not to be confused with maven profiles). This profile executes the bash function [perf_tests](https://github.com/jbosstm/narayana/blob/master/scripts/hudson/narayana.sh#L933) in the narayana repo. The actual benchmarks are in a separate repo (take a look at what the linked perf_tests bash function does) and these are executed by a bash function called [run_benchmarks](https://github.com/jbosstm/performance/blob/master/scripts/run_bm.sh#L41)

So you will need to modify the list of tests executed by that function (run_benchmarks). I would recommend just running a single test for now which starts a transaction, enlists a dummy resource and then commits. Do any of the existing tests do that, namely:

``` bash
BM1="com.hp.mwtests.ts.arjuna.performance.Performance1.*"
BM2="com.hp.mwtests.ts.arjuna.atomicaction.CheckedActionTest.*"
BM3="com.arjuna.ats.jta.xa.performance.JTAStoreTests.*"
```

Notice that this approach also means that we do not require any pom changes.

Reply:
I have included BM1 and BM3 as these test cases does what you mentioned.
Okay I have made the changes as per your request. I have raised separate PR for Narayana repo changes.